tools/fragpipe/fragpipe.xml

```diff
@@ -0,0 +1,286 @@
+                philosopher workspace --init &&
+                '$__tool_directory__/genericize_db.py' ${database_name} genericized_db.fas &&
+                philosopher database --custom 'genericized_db.fas' --prefix rev_
+                #if str($database_options) != 'decoy_and_contam'
+                    --contam
+                #end if
+                --contamprefix &&
+                mv *-decoys-*.fas decoys-contam-database.fas &&
+            #end if
+            fragpipe --msfragger_key f28c4be2-759b0a11-6c1ea652-1240565a --ionquant_key 47cb2953-b4bf7726-2c96f5fe-8f1013aa --headless --threads $threads --ram \$[ \${GALAXY_MEMORY_MB:-8192} / 1024 ] --workflow fp.workflow --manifest fp.manifest --workdir outputs
+            --config-msfragger \${CONDA_PREFIX}/share/msfragger-*/MSFragger-*/MSFragger-*.jar --config-philosopher `which philosopher` --config-python `which python` --config-ionquant \${CONDA_PREFIX}/share/ionquant-*/IonQuant.jar
+            && cd outputs
+            && ln -s log_*.txt log.txt
+            #if $output_options and 'combined_outputs' in $output_options
+                && mkdir combined_outputs
+                && for f in combined*tsv; do [[ -e $\{f} || continue ]]; mv \${f} combined_outputs; done
+            #end if
+            && for f in psm.tsv ion.tsv peptide.tsv protein.tsv; do n=0; for i in `find * -name \${f}`; do d=\${i%/*}; echo "\${n}  \${d} \${i}"; if [[ \$n == 0 ]]; then awk 'NR==1{printf("Experiment\t%s\n",\$0);}' \$i > concat_\${f}; fi; awk 'NR>1{printf("'"\${d}"'\t%s\n",\$0);}' \$i >> concat_\${f}; t=`echo \${i} | tr '/' '.'`; mv \${i} \${t}; n=\$((n+1)); done; done
+        #end if
+    ]]></command>
+
+    <!-- Config file is a FragPipe workflow file -->
+    <configfiles>
+        <expand macro="workflow_configfile" />
+    </configfiles>
+
+    <inputs>
+        <!-- License agreements -->
+        <expand macro="license_agreements"/>
+
+        <!-- Input scan samples -->
+        <expand macro="samples"/>
+
+        <!-- Searched database -->
+        <expand macro="fragger_database"/>
+
+        <!-- Workflow input, if the user chooses to re-use existing FragPipe workflow. -->
+        <expand macro="workflow"/>
+
+        <!-- Only generate a workflow, don't run FragPipe -->
+        <param name="workflow_only" type="boolean" truevalue="yes" falsevalue="no" checked="false" label="Only generate the workflow output"/>
+
+        <param name="output_options" label="Additional outputs" type="select" multiple="true" display="checkboxes" optional="true">
+            <option value="workflow">FragPipe Workflow</option>
+            <option value="log">FragPipe Log</option>
+            <option value="combined_outputs">FragPipe Combined Outputs</option>
+            <option value="concatenated_outputs">TSV Outputs Concatenated</option>
+        </param>
+
+        <param argument="--threads" type="integer" value="8" label="Number of threads"/>
```

Comment:
This should not be controllable by the user. Instead use `\${GALAXY_SLOTS:-8}` in the command.

Reply:
Fixed to use GALAXY_SLOTS.